src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```
@@ -91,41 +115,65 @@ export class PdfPreviewComponent implements OnInit, OnDestroy {
 
-            for (let i = 1; i <= pdf.numPages; i++) {
+            for (let i = 1; i <= this.totalPages; i++) {
                 const page = await pdf.getPage(i);
                 const viewport = page.getViewport({ scale: 2 });
-                const canvas = this.createCanvas(viewport);
+                const canvas = this.createCanvas(viewport, i);
                 const context = canvas.getContext('2d');
-                if (context) {
-                    await page.render({ canvasContext: context, viewport }).promise;
-                }
+                await page.render({ canvasContext: context!, viewport }).promise;
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)